packages/kbn-dev-utils/src/code_ownership/record.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { PathLike, writeFileSync } from 'fs';
+import { OwnershipRule } from './ownership_config';
+import { ToolingLog } from '../tooling_log';
+
+type FileList = string[];
+type ExcludedFileList = string[];
+
+const appendUtf8 = { flag: 'a', encoding: 'utf8' };
+
+const write = (
+  output: PathLike,
+  owner: string,
+  files: FileList,
+  excludeFiles: ExcludedFileList
+) => {
+  const writeToFile = writeFileSync.bind(null, output);
+  writeToFile(`\n#${drop(owner)}\n`, appendUtf8);
+
+  for (const file of files) {
+    writeToFile(`${file} ${owner}\n`, appendUtf8);
+  }
+
+  for (const excluded of excludeFiles) {
+    const formatted = `!${excluded}`;
+    writeToFile(`${formatted} ${owner}\n`, appendUtf8);
+  }
+};
+
+export const record = (output: string, log: ToolingLog, rules: OwnershipRule[]) => {
+  writeFileSync(output, preamble(), { encoding: 'utf8' });
+
+  for (const { files, excludeFiles, codeOwner } of rules) {
+    if (codeOwner) {
+      write(output as PathLike, codeOwner, files, excludeFiles);
+    }
+  }
+
+  log.info(`\n### CODEOWNERS generation complete against: ${output}`);
+};
+
+function drop(x: string) {
+  return x.replace(/@/, '').replace(/elastic\//, '');
+}
+
+function preamble() {
+  return `# GENERATED by @kbn/dev-utils (Do not edit manually)
```

Comment:
Let's add a bit of text about where to go to edit the config and how to regenerate it.

Reply:
+1, I actually added this coment to the output and realize you already mentioned it here with is more accurate.